librz/analysis/rzil/rzil.c

```diff
@@ -0,0 +1,144 @@
+// SPDX-FileCopyrightText: 2021 heersin <teablearcher@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_analysis.h>
+
+/**
+ * Create an empty RzAnalysisRzil instance
+ * inner VM should be init in adaptive plugin
+ * \return RzAnalysisRzil* a pointer to RzAnalysisRzil instance
+ */
+RZ_API RzAnalysisRzil *rz_analysis_rzil_new() {
+	RzAnalysisRzil *rzil = RZ_NEW0(RzAnalysisRzil);
+	if (!rzil) {
+		return NULL;
+	}
+	rzil->vm = RZ_NEW0(struct rz_il_vm_t);
+	if (!rzil->vm) {
+		free(rzil);
+		return NULL;
+	}
+	return rzil;
+}
+
+/**
+ * Cleanup RZIL instance : clean VM, clean arch-specific user_data, and RZIL itself
+ * \param analysis pointer to rizin's RzAnalysis
+ * \param rzil pointer to RzAnalysisRzil
+ */
+RZ_API void rz_analysis_rzil_cleanup(RzAnalysis *analysis, RzAnalysisRzil *rzil) {
+	if (!rzil) {
+		return;
+	}
+	if (analysis && analysis->cur && analysis->cur->rzil_fini) {
+		analysis->cur->rzil_fini(analysis);
+	}
+	free(rzil);
+	analysis->rzil = NULL;
+}
+
+/**
+ * Set instruction pointer for the current RZIL VM session
+ * \param rzil RzAnalysis* pointer to RzAnalysisRzil instance
+ * \param addr ut64 address of new pc
+ * \return true if set successfully, else return false
+ */
+RZ_API bool rz_analysis_rzil_set_pc(RzAnalysisRzil *rzil, ut64 addr) {
+	if (!rzil) {
+		return false;
+	}
+	rzil->pc_addr = addr;
+	return true;
+}
+
+/**
+ * Init an empty RZIL
+ * \param analysis RzAnalysis* pointer to RzAnalysis
+ * \param rzil RzAnalysisRzil* pointer to RzAnalysisRzil
+ * \param romem int is read only mem ?
+ * \param stats int use stats ?
+ * \param nonull int is pc cannot be null ?
+ * \return true if setup, else return false
+ */
+RZ_API bool rz_analysis_rzil_setup(RzAnalysis *analysis) {
```

Comment:
Is this expected to be call only after rz_analysis_rzil_cleanup? If so, add a `rz_return_val_if_fail(!analysis->rzil, false)` to make sure that we don't leak the previous value of `analysis->rzil`.